### PR TITLE
KOGITO-1924 DMN DMNDecisionModel to expose via Kogito API DMNModel

### PR DESCRIPTION
**Pull request by tarilabs**

### CODE REVIEW

File: drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DmnDecisionModel.java [hunk 1]
```diff
@@ -26,19 +26,22 @@ public DMNContext newContext(Map<String, Object> variables) {
...
         return dmnRuntime.evaluateAll(dmnModel, context);
     }
 
     @Override
     public DMNResult evaluateDecisionService(DMNContext context, String decisionServiceName) {
+        DMNModel dmnModel = getDMNModel();
+        return dmnRuntime.evaluateDecisionService(dmnModel, context, decisionServiceName);
+    }
+
+    public DMNModel getDMNModel() {
```
Comment (tarilabs) [inline drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DmnDecisionModel.java:39]: my preference would be to `@Override` this so it is available up in Kogito API's `DecisionModel`
do you concur @mswiderski ? :)

Comment (mswiderski) [reply drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DmnDecisionModel.java:39]: +1 to expose it on `DecisionModel` if possible

Comment (tarilabs) [reply drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DmnDecisionModel.java:40]: done with 49dbda4

File: drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DmnDecisionModel.java [hunk 2]
```diff
@@ -26,19 +26,23 @@ public DMNContext newContext(Map<String, Object> variables) {
...
+
+    @Override
+    public DMNModel getDMNModel() {
         DMNModel dmnModel = dmnRuntime.getModel(namespace, name);
         if (dmnModel == null) {
-            throw new IllegalArgumentException("DMN model '" + name + "' not found with namespace '" + namespace + "'");
+            throw new IllegalStateException("DMN model '" + name + "' not found with namespace '" + namespace + "' in the inherent DMNRuntime.");
         }
-        return dmnRuntime.evaluateDecisionService(dmnModel, context, decisionServiceName);
+        return dmnModel;
```
Comment (danielezonca) [inline drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DmnDecisionModel.java:45]: I would prefer to do it in the constructor and store it as field. Is there any reason to keep this `get` lazy?

**Comment by tarilabs** [comment]
```
[INFO] SpringBoot Kogito Archetype ........................ FAILURE [ 45.696 s]

[INFO] Quarkus Kogito Archetype ........................... FAILURE [  9.991 s]

[INFO] ------------------------------------------------------------------------

[INFO] BUILD FAILURE

[INFO] ------------------------------------------------------------------------
```

Jenkins test again

**Comment by tarilabs** [comment]
@mswiderski glad to hear it, I have the other modification in the list of active work tasks, will keep posted when I have some progress there too

**Comment by tarilabs** [comment]
Jenkins test again

**Comment by tarilabs** [comment]
Jenkins test again

**Pull request merged**